src/Language.php

```diff
@@ -819,9 +821,7 @@ public function getUsed()
 	 */
 	public function hasKey($string)
 	{
-		$key = strtoupper($string);
```

Comment:
You're loosing the `strtoupper` here - we should probably sanitise this in the `MessageCatalogue` somehow

Reply:
I don't think there is a reason we need to enforce uppercase keys personally.  Especially because they aren't going into storage uppercased, only searched that way.

Reply:
We don't need to enforce it - but it seems sensible to be case insensitive - whatever that practically means.

Reply:
Storage can look like this:

```php
$store = [
    'foo' => 'bar',
    'Foo' => 'bar',
    'FOO' => 'bar',
];
```

So standardizing on lookup only makes little sense.  We should be standardizing the keys at both storage and lookup, or in neither spot.  If there's a valid reason to enforce all caps keys, we can do it, but let's be consistent about it; right now we aren't.

Reply:
OK In that case let's standardise in storage

```
$store = [
    'foo' => 'bar',
    'Foo' => 'something',
    'FOO' => 'another_thing',
];
```

seems like we are going to set ourselves up for disaster. At least before even if all those things existed in a config file only one of them ever got loaded.

Reply:
Actually, it's fine as is.  https://3v4l.org/uaY8o

So, our behavior actually keeps you from accessing `$store['foo']` and `$store['Foo']` right now, no matter what you give only `$store['FOO']` would be returned.

So if anything, document the internal behavior change and be done with it.  Make your use of language strings match the declaring case in your source code.

Reply:
No i get that now all these are possible to store, and now can actually be accessed and I'm asserting that's not a good thing (in my opinion). I would document an internal behaviour change and `strtoupper` on storage and retrieval and be formally case insensitive rather than semi-case insensitive

Reply:
Doesn't seem there's a performance hit to do that with array_change_key_case() - https://3v4l.org/IdIFY - I'll do it then.

Reply:
We need to `strtoupper` the request key still. See https://3v4l.org/nOZ0D

Reply:
That'll all be self contained in the catalogue.  Language class won't know or care about the catalogue's internals.

Reply:
Rebased.  There is one caveat, somewhat hinted at in the upgrade doc.  When a fallback catalogue is defined, the key will not be passed to it standardized to uppercase.  This means it is still possible to create a catalogue subclass without enforcing key standardization if you truly want that type of behavior; either way our API doesn't care (as should be the case).